spec/param_type_mismatch_spec.rb

```diff
@@ -0,0 +1,44 @@
+require_relative 'shared'
+
+describe 'parameter_type_mismatch1.sag' do
+  include_context 'component test', 'fixtures/parameter_type_mismatch1.sag'
+
+  include_examples 'parsing', ASTree.new(
+    Program.new(
+      Function.new(
+        'main',
+        { [] => :INT },
+        Clause.new(
+          nil,
+          Return.new(
+            Expression.new(
+              :test,
+              IntegerConstant.new(3),
+              StringConstant.new('hello'))))),
+      Function.new(
+        'test',
+        { %i[INT INT] => :INT },
+        Clause.new(
+          Parameter.new(:X),
+          Parameter.new(:Y),
+          nil,
+          Return.new(
+            Expression.new(
+              :+,
+              Variable.new(:X),
+              Variable.new(:Y)))))))
+
+  include_examples 'validation error', <<~ERROR
+    test expects 2 parameters: INT, INT
```

Comment:
make the function name clearer that it is the function name and not just a word in the error message

Reply:
```suggestion
    function ':test' expects 2 parameters: INT, INT
```